octavia_proxy/api/drivers/utils.py

```diff
@@ -67,3 +68,66 @@ def call_provider(provider, driver_method, *args, **kwargs):
         LOG.exception("Provider '%s' raised an unknown error: %s",
                       provider, str(e))
         raise exceptions.ProviderDriverError(prov=provider, user_msg=e)
+
+def _base_to_provider_dict(current_dict, include_project_id=False):
+    new_dict = copy.deepcopy(current_dict)
+    if 'provisioning_status' in new_dict:
```

Comment:
now that I see the code I wonder how it came to that
`new_dict.pop('XXX', None)` does the trick not requiring this if. But in any way - we need to think whether this is really required to be done this way. We decided that driver returns proper response object. On the other side we pass to into the driver also request object.

Reply:
new_dict.pop("key", None)

Reply:
@kucerakk make a list for keys, and use list for pop this params if exist